python/nutpie/compile_pymc.py

```diff
@@ -231,7 +231,7 @@ def _compute_shapes(model):
         inputs=[],
         outputs=[var.shape for var in trace_vars.values()],
         givens=(
-            [(obs, obs.tag.observations) for obs in model.observed_RVs]
+            [(obs, model.rvs_to_values[obs]) for obs in model.observed_RVs]
```

Comment:
This is not even needed. `observed_RVs` have the same shape as their observations

Reply:
Should I ignore this?

Reply:
I have example models where other RVs depend on observed RVs

Reply:
That's not a problem. By definition rv.shape must match observations.shape. Otherwise it's an invalid PyMC model.

No strong preference. I think this was copied from PyMC when that relationship was not assumed / required.